[BUG] ClusterManagerTaskThrottlingIT.testTimeoutWhileThrottling is flaky

Observed a couple times such as: https://build.ci.opensearch.org/job/gradle-check/6359/

Can usually be reproduced with the following: 

```
./gradlew ':server:internalClusterTest' -Dtests.iters=30 --tests "org.opensearch.clustermanager.ClusterManagerTaskThrottlingIT"
```

Example failure:

```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.clustermanager.ClusterManagerTaskThrottlingIT.testTimeoutWhileThrottling {seed=[F40BF37A733FFE89:6AF0FE6F91F21DCC]}" -Dtests.seed=F40BF37A733FFE89 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=fr-BE -Dtests.timezone=Asia/Tel_Aviv -Druntime.java=17

org.opensearch.clustermanager.ClusterManagerTaskThrottlingIT > testTimeoutWhileThrottling {seed=[F40BF37A733FFE89:6AF0FE6F91F21DCC]} FAILED
    java.lang.AssertionError: expected:<1> but was:<0>
        at __randomizedtesting.SeedInfo.seed([F40BF37A733FFE89:6AF0FE6F91F21DCC]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.opensearch.clustermanager.ClusterManagerTaskThrottlingIT.testTimeoutWhileThrottling(ClusterManagerTaskThrottlingIT.java:175)
```

--- a/server/src/internalClusterTest/java/org/opensearch/clustermanager/ClusterManagerTaskThrottlingIT.java
+++ b/server/src/internalClusterTest/java/org/opensearch/clustermanager/ClusterManagerTaskThrottlingIT.java
@@ -164,9 +164,9 @@ public class ClusterManagerTaskThrottlingIT extends OpenSearchIntegTestCase {
 
                 @Override
                 public void onFailure(Exception e) {
+                    timedoutRequest.incrementAndGet();
                     latch.countDown();
                     assertTrue(e instanceof ProcessClusterEventTimeoutException);
-                    timedoutRequest.incrementAndGet();
                 }
             };
             executePutMappingRequests(totalRequest, node, listener);

